### PR TITLE
Fix most depwarns and errors on 0.7

### PR DESCRIPTION
**Pull request by yuyichao**

### CODE REVIEW

File: src/multivariate/solvers/zeroth_order/nelder_mead.jl [hunk 1]
```diff
@@ -227,7 +227,7 @@ function update_state!(f::F, state::NelderMeadState{T}, method::NelderMead) wher
         else # f_reflect > f_highest
             # Inside constraction
             @simd for j in 1:n
-                @inbounds state.x_cache[j] = state.x_centroid[j] - γ *(state.x_reflect[j] - state.x_centroid[j])
+                @inbounds state.x_cache[j] = state.x_centroid[j] - state.γ *(state.x_reflect[j] - state.x_centroid[j])
```
Comment (yuyichao) [inline src/multivariate/solvers/zeroth_order/nelder_mead.jl]: This is a behavioral change but I think it's correct from the context...

Comment (pkofod) [reply src/multivariate/solvers/zeroth_order/nelder_mead.jl]: Good catch... I really wish these unicode irrationals had to be explicitly loaded.

**Comment by pkofod** [comment]
I'll merge this when metadata is sorted out for nlsolversbase...

**Comment by yuyichao** [comment]
Ah, that explains why this package haven't been updated for it.... I just pull from the latest master/tag and didn't pay attention to metadata....

**Comment by pkofod** [comment]
> Ah, that explains why this package haven't been updated for it.... I just pull from the latest master/tag and didn't pay attention to metadata....

It's fine... We're not tagging before that is merged anyway, so.. But thanks for fixing these!

**Comment by pkofod** [comment]
I'm not going to wait for mac tests on travis as they might not fire this year.... Thanks for doing this @yuyichao

**Pull request merged**